File: backend/internal/modules/admin/content/dto/content.go
// Package dto 定义内容中心的出入参结构（管理端）。
package dto

// ArticleListQuery 文章列表查询。
type ArticleListQuery struct {
	Kind       string `form:"kind"`        // news/help/terms/privacy；空为全部
	CategoryID uint64 `form:"category_id"` // 分类筛选；0 为不限
	Status     string `form:"status"`      // draft/published/offline；空为全部
	Keyword    string `form:"keyword"`     // 标题模糊搜索
	Page       int    `form:"page"`
	PageSize   int    `form:"page_size"`
}

// ArticleSaveRequest 创建 / 更新文章。
//
// Body 走服务端净化（internal/pkg/sanitize），前端提交什么都会被白名单过滤一遍再入库。
type ArticleSaveRequest struct {
	Kind       string `json:"kind" binding:"required"`
	CategoryID uint64 `json:"category_id"`
	Slug       string `json:"slug"`
	Title      string `json:"title" binding:"required"`
	Summary    string `json:"summary"`
	Body       string `json:"body"`
	BodyFormat string `json:"body_format"`
	Cover      string `json:"cover"`
	Tags       string `json:"tags"`
	Pinned     *bool  `json:"pinned"`
	SortOrder  int    `json:"sort_order"`
	Version    string `json:"version"`
	PublishAt  string `json:"publish_at"` // 2006-01-02 15:04:05；空=立即发布
}

// ArticleInfo 文章信息。
type ArticleInfo struct {
	ID           uint64 `json:"id"`
	Kind         string `json:"kind"`
	CategoryID   uint64 `json:"category_id"`
	CategoryName string `json:"category_name"`
	Slug         string `json:"slug"`
	Title        string `json:"title"`
	Summary      string `json:"summary"`
	Body         string `json:"body"`
	BodyFormat   string `json:"body_format"`
	Cover        string `json:"cover"`
	Tags         string `json:"tags"`
	Status       string `json:"status"`
	Pinned       bool   `json:"pinned"`
	SortOrder    int    `json:"sort_order"`
	Version      string `json:"version"`
	PublishAt    string `json:"publish_at"`
	OfflineAt    string `json:"offline_at"`
	OperatorID   uint64 `json:"operator_id"`
	CreatedAt    string `json:"created_at"`
	UpdatedAt    string `json:"updated_at"`
}

// ArticleListResponse 文章分页列表。
type ArticleListResponse struct {
	Total    int64          `json:"total"`
	List     []*ArticleInfo `json:"list"`
	Page     int            `json:"page"`
	PageSize int            `json:"page_size"`
}

// CategoryListQuery 分类查询。
type CategoryListQuery struct {
	Kind   string `form:"kind"`   // news/help；空为全部
	Status string `form:"status"` // active/disabled；空为全部（保证树完整）
}

// CategorySaveRequest 创建 / 更新分类。
type CategorySaveRequest struct {
	Kind        string `json:"kind" binding:"required"`
	ParentID    uint64 `json:"parent_id"`
	Name        string `json:"name" binding:"required"`
	Slug        string `json:"slug"`
	Description string `json:"description"`
	Icon        string `json:"icon"`
	SortOrder   int    `json:"sort_order"`
	Status      string `json:"status"`
}

// CategoryInfo 分类信息（含子节点树）。
type CategoryInfo struct {
	ID          uint64          `json:"id"`
	Kind        string          `json:"kind"`
	ParentID    uint64          `json:"parent_id"`
	Name        string          `json:"name"`
	Slug        string          `json:"slug"`
	Description string          `json:"description"`
	Icon        string          `json:"icon"`
	SortOrder   int             `json:"sort_order"`
	Status      string          `json:"status"`
	Children    []*CategoryInfo `json:"children,omitempty"`
}

// CategoryListResponse 分类树响应。
type CategoryListResponse struct {
	Items []*CategoryInfo `json:"items"`
}

// LinkListQuery 友情链接查询。
type LinkListQuery struct {
	Status   string `form:"status"`
	Keyword  string `form:"keyword"`
	Page     int    `form:"page"`
	PageSize int    `form:"page_size"`
}

// LinkSaveRequest 创建 / 更新友情链接。
type LinkSaveRequest struct {
	Name        string `json:"name" binding:"required"`
	URL         string `json:"url" binding:"required"`
	Logo        string `json:"logo"`
	Description string `json:"description"`
	SortOrder   int    `json:"sort_order"`
	OpenInNew   *bool  `json:"open_in_new"`
	Status      string `json:"status"`
}

// LinkInfo 友情链接信息。
type LinkInfo struct {
	ID          uint64 `json:"id"`
	Name        string `json:"name"`
	URL         string `json:"url"`
	Logo        string `json:"logo"`
	Description string `json:"description"`
	SortOrder   int    `json:"sort_order"`
	OpenInNew   bool   `json:"open_in_new"`
	Status      string `json:"status"`
	CreatedAt   string `json:"created_at"`
	UpdatedAt   string `json:"updated_at"`
}

// LinkListResponse 友情链接分页列表。
type LinkListResponse struct {
	Total    int64       `json:"total"`
	List     []*LinkInfo `json:"list"`
	Page     int         `json:"page"`
	PageSize int         `json:"page_size"`
}

File: frontend-admin/src/api/content.ts
// 内容中心（doc100）：文章（新闻/帮助/条款/隐私）/ 分类 / 友情链接。
//
// 与公告 API（@/api/notification.ts）分开的原因：公告带「平台定向 / 等级 / 弹窗」的
// 站内消息语义，走 notify:* 权限；这里是纯展示型内容，走 content:* 权限。
import { request } from '@/utils/request'

export type ArticleKind = 'news' | 'help' | 'terms' | 'privacy'
export type ArticleStatus = 'draft' | 'published' | 'offline'
export type CategoryStatus = 'active' | 'disabled'

export interface ArticleItem {
  id: number
  kind: ArticleKind
  category_id: number
  category_name: string
  slug: string
  title: string
  summary: string
  body: string
  body_format: 'html' | 'text'
  cover: string
  tags: string
  status: ArticleStatus
  pinned: boolean
  sort_order: number
  version: string
  publish_at: string
  offline_at: string
  operator_id: number
  created_at: string
  updated_at: string
}

export interface ArticleListResponse {
  total: number
  list: ArticleItem[]
  page: number
  page_size: number
}

export interface ArticleListQuery {
  kind?: string
  category_id?: number
  status?: string
  keyword?: string
  page?: number
  page_size?: number
  // request.get 的 params 形参是 Record<string, unknown>，命名接口不带隐式索引签名，
  // 这里显式声明（与 api/logcenter.ts 的 LogQueryParams 同做法）才能直接传对象。
  [key: string]: string | number | undefined
}

export interface ArticleSaveRequest {
  kind: ArticleKind
  category_id?: number
  slug?: string
  title: string
  summary?: string
  body: string
  body_format?: 'html' | 'text'
  cover?: string
  tags?: string
  pinned?: boolean
  sort_order?: number
  version?: string
  publish_at?: string
}

export interface CategoryItem {
  id: number
  kind: 'news' | 'help'
  parent_id: number
  name: string
  slug: string
  description: string
  icon: string
  sort_order: number
  status: CategoryStatus
  children?: CategoryItem[]
}

export interface CategorySaveRequest {
  kind: 'news' | 'help'
  parent_id?: number
  name: string
  slug?: string
  description?: string
  icon?: string
  sort_order?: number
  status?: CategoryStatus
}

export interface FriendlyLinkItem {
  id: number
  name: string
  url: string
  logo: string
  description: string
  sort_order: number
  open_in_new: boolean
  status: CategoryStatus
  created_at: string
  updated_at: string
}

export interface FriendlyLinkListResponse {
  total: number
  list: FriendlyLinkItem[]
  page: number
  page_size: number
}

export interface LinkSaveRequest {
  name: string
  url: string
  logo?: string
  description?: string
  sort_order?: number
  open_in_new?: boolean
  status?: CategoryStatus
}

// ---------- 文章 ----------

export function getArticles(params: ArticleListQuery): Promise<ArticleListResponse> {
  return request.get<ArticleListResponse>({ url: '/content/articles', params })
}

export function getArticle(id: number): Promise<ArticleItem> {
  return request.get<ArticleItem>({ url: `/content/articles/${id}` })
}

export function createArticle(data: ArticleSaveRequest): Promise<ArticleItem> {
  return request.post<ArticleItem>({ url: '/content/articles', data })
}

export function updateArticle(id: number, data: ArticleSaveRequest): Promise<ArticleItem> {
  return request.put<ArticleItem>({ url: `/content/articles/${id}`, data })
}

export function publishArticle(id: number): Promise<ArticleItem> {
  return request.post<ArticleItem>({ url: `/content/articles/${id}/publish` })
}

export function offlineArticle(id: number): Promise<ArticleItem> {
  return request.post<ArticleItem>({ url: `/content/articles/${id}/offline` })
}

export function deleteArticle(id: number): Promise<string> {
  return request.delete<string>({ url: `/content/articles/${id}` })
}

// ---------- 分类 ----------

export interface CategoryListResponse {
  items: CategoryItem[]
}

export function getContentCategories(params: { kind?: string; status?: string }): Promise<CategoryListResponse> {
  return request.get<CategoryListResponse>({ url: '/content/categories', params })
}

export function createContentCategory(data: CategorySaveRequest): Promise<CategoryItem> {
  return request.post<CategoryItem>({ url: '/content/categories', data })
}

export function updateContentCategory(id: number, data: CategorySaveRequest): Promise<CategoryItem> {
  return request.put<CategoryItem>({ url: `/content/categories/${id}`, data })
}

export function deleteContentCategory(id: number): Promise<string> {
  return request.delete<string>({ url: `/content/categories/${id}` })
}

// ---------- 友情链接 ----------

export function getFriendlyLinks(params: {
  status?: string
  keyword?: string
  page?: number
  page_size?: number
}): Promise<FriendlyLinkListResponse> {
  return request.get<FriendlyLinkListResponse>({ url: '/content/links', params })
}

export function createFriendlyLink(data: LinkSaveRequest): Promise<FriendlyLinkItem> {
  return request.post<FriendlyLinkItem>({ url: '/content/links', data })
}

export function updateFriendlyLink(id: number, data: LinkSaveRequest): Promise<FriendlyLinkItem> {
  return request.put<FriendlyLinkItem>({ url: `/content/links/${id}`, data })
}

export function deleteFriendlyLink(id: number): Promise<string> {
  return request.delete<string>({ url: `/content/links/${id}` })
}
